src/register_users/shared_secret_register/v1.rs

```diff
@@ -76,3 +83,37 @@ impl Response {
         Self { access_token, user_id, home_server, device_id }
     }
 }
+
+/// Calculate the MAC based on the given inputs.
+///
+/// See <https://matrix-org.github.io/synapse/latest/admin_api/register_api.html#shared-secret-registration> for details.
+#[cfg(feature = "shared-secret-registration-mac")]
+pub fn hmac(
+    registration_shared_secret: &[u8],
+    nonce: &[u8],
+    username: &[u8],
+    password: &[u8],
```

Comment:
Since these are strings elsewhere in the API, I think `&str` makes more sense here.